store/store.go

```diff
@@ -258,12 +258,13 @@ func (bs *BlockStore) LoadBlockCommit(height int64) *types.Commit {
 	return commit
 }
 
-// LoadSeenCommit returns the locally seen Commit for the given height.
-// This is useful when we've seen a commit, but there has not yet been
-// a new block at `height + 1` that includes this commit in its block.LastCommit.
-func (bs *BlockStore) LoadSeenCommit(height int64) *types.Commit {
+// LoadSeenCommit returns the last locally seen Commit before being
+// cannonicalized. This is useful when we've seen a commit, but there
+// has not yet been a new block at `height + 1` that includes this
+// commit in its block.LastCommit.
+func (bs *BlockStore) LoadSeenCommit() *types.Commit {
```

Comment:
Looks like the semantics of this method have changed slightly. It no longer loads a Seen commit for any given height, but instead loads the _last_ seen commit. I would advocate for changing the name to `LoadLastSeenCommit`.

Reply:
To provide some back story to this. We used to save all seen commits at every height. But I realised that after the commit became canonicalized we never needed the seen commit again. We only persisted it in case the proposer of the following height crashed so that they could recover the commit they had seen and add that to their proposed block. I've opted to keeping the term `LoadSeenCommit` as to indicate that there is only ever one seen commit at a time. I admit the comment you've pointed out doesn't quite reflect that. I'm open to suggestions about how to change the semantics to best represent that.